src/components/RepoSearchForm.tsx

```diff
@@ -8,17 +8,29 @@ import { GithubRepositoryItem } from '../modules/github/types';
 import './RepoSearchForm.css';
 import Search from 'antd/lib/input/Search';
 
+const INITIAL_PAGE_SIZE = 10;
+const MAX_PAGES_AMOUNT = 1000;
+
 export const GithubSearchRepositories: React.FC = () => {
     const dispatch: any = useDispatch();
+    const {list, isLoading, totalRepositoriesCount} = useSelector((state: StoreState) => state.gitHubRepositories);
     const [searchName, setSearchName] = useState('');
-    const { list, isLoading } = useSelector((state: StoreState) => state.gitHubRepositories);
+    const [pageSize, setPageSize] = useState<number>(INITIAL_PAGE_SIZE);
```

Comment:
Let's move pagination state to `gitHubRepositories` reducer, so that we have a single source of truth regarding repositories state.
```typescript
export interface GithubRepositoriesState {
    ...current fields
    pagination: {
        page: number;
        size: number;
    };
}
```